dapr-spring/dapr-spring-boot-tests/src/main/java/io/dapr/spring/boot/tests/DaprContainerConnectionDetailsFactory.java

```diff
@@ -0,0 +1,45 @@
+package io.dapr.spring.boot.tests;
+
+import io.dapr.spring.boot.autoconfigure.client.DaprConnectionDetails;
+import io.dapr.testcontainers.DaprContainer;
+import org.springframework.boot.testcontainers.service.connection.ContainerConnectionDetailsFactory;
+import org.springframework.boot.testcontainers.service.connection.ContainerConnectionSource;
+
+public class DaprContainerConnectionDetailsFactory
```

Comment:
It should also be registered at `src/main/resources/META-INF/spring.factories`

```
org.springframework.boot.autoconfigure.service.connection.ConnectionDetailsFactory=\
io.dapr.spring.boot.tests.DaprContainerConnectionDetailsFactory
```

```suggestion
class DaprContainerConnectionDetailsFactory
```